contracts/root/depositManager/DepositManager.sol

```diff
@@ -56,4 +59,36 @@ contract DepositManager is DepositManagerStorage, IDepositManager {
   {
     rootChain.createDepositBlock(_user, _token, amountOrNFTId);
   }
+
+  function tokenFallback(address _user, uint256 _amount, bytes _data) public {
```

Comment:
When will this function be invoked?

Reply:
erc233 safeTransferFrom(so it will be better UX for deposit) and same for OnERC721receiver.